molecule/templates/vagrantfile.j2

```diff
@@ -110,7 +133,9 @@ Vagrant.configure('2') do |config|
 
   {% for instance in config.vagrant.instances -%}
   config.vm.define '{{ instance.vm_name }}' do |c|
+    {%- if 'libvirt' not in config.vagrant.providers|map(attribute='type')|list %}
     c.vm.hostname = '{{ instance.vm_name }}'
+    {%- endif %}
```

Comment:
This conditional is a big ugly, but required for automatic box creation under libvirt, since vagrant-libvirt does not support setting hostnames. See here: https://github.com/vagrant-libvirt/vagrant-libvirt/issues/289